pandas/core/indexing.py

```diff
@@ -1925,7 +1925,7 @@ def maybe_droplevels(index, key):
     return index
 
 
-def _non_reducing_slice(slice_):
+def _non_reducing_slice(slice_, axis=0):
```

Comment:
prob also should pass ``self.data`` here, mainly for error checking / ndim (though this really only works for a DataFrame).

@TomAugspurger